### ISSUE TITLE
Create AppArmor profile for the operator

### ISSUE THREAD

**Issue by pjbgf** [kubernetes-sigs/security-profiles-operator#65]
#### What would you like to be added:
A tailor-made AppArmor profile to restrict functionality to the minimum required for the operator to run.

#### Why is this needed:
Better define the attack surface, giving users a clear view of what the operator can/cannot do.

/area security
/priority important-longterm

**Comment by saschagrunert** [comment]
I love that! Would the deployment work in the same way like we do in #52?

Since the complexity of the init container seems to increase we could think about wrapping it in an own container/binary. 😅

**Comment by pjbgf** [comment]
Yes, we probably would have to deploy it in the same way as #52. 

As for its own binary, there are a few issues as we discussed off-line. We could isolate each one of those "self profile deployment" in specific init containers, that could make it easier for easier to opt-out.

**Comment by saschagrunert** [comment]
@pjbgf do you have any update on this one?

**Comment by pjbgf** [comment]
@saschagrunert I should get something ready this week.

**Comment by pjbgf** [comment]
@saschagrunert we may need to think on a different approach to deploy the operator's own apparmor profile, as I mentioned on the PR and on the k8s issue (kubernetes/kubernetes#97273), the approach taken on seccomp does not work for AppArmor.

**Comment by saschagrunert** [comment]
> @saschagrunert we may need to think on a different approach to deploy the operator's own apparmor profile, as I mentioned on the PR and on the k8s issue ([kubernetes/kubernetes#97273](https://github.com/kubernetes/kubernetes/issues/97273)), the approach taken on seccomp does not work for AppArmor.

Sounds good, let's put it on the agenda for the community meeting this week. :+1:

**Comment by pjbgf** [comment]
The issue upstream (https://github.com/kubernetes/kubernetes/issues/97273) has now been fixed (thanks to @saschagrunert), which is yet to be released on a new version.
A non "kube-native" approach may need to be taken to sandbox SPO's components that would work for all currently supported Kubernetes versions.

**Comment by saschagrunert** [comment]
@pjbgf I assume we prefer the kube-native way? Would it be possible to apply the profile only on specific kubernetes versions by  checking it before deploying the daemon?

**Comment by pjbgf** [comment]
> @pjbgf I assume we prefer the kube-native way? Would it be possible to apply the profile only on specific kubernetes versions by checking it before deploying the daemon?

@saschagrunert yes, I think we are better off doing kube-native, which in this case would mean deploying the profile in a container init.
Now I am not sure how we would check the Kubernetes version. In this specific case we probably would want to know the kubelet version.

**Comment by saschagrunert** [comment]
> Potentially the daemonset could just check "is my AppAmor profile loaded? If so, use it, otherwise fall-back to default".

Yes I think this would make sense :+1:

**Comment by ccojocar** [comment]
I think now we can use the apparmor in-cluster profile recorder to create/maintain a profile.
